File: code/platform-server/park-enterprise/src/main/java/com/cloudhub/platform/enterprise/domain/entity/Enterprise.java
package com.cloudhub.platform.enterprise.domain.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.cloudhub.platform.common.domain.BaseEntity;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 企业档案 (park-enterprise 核心实体)
 *
 * <p>csyh 业务融合 Phase 1: Enterprise 字段基于 csyh {@code pai-enterprise-csyh-2.x}
 *    EnterprisePO 重写 (60+ 字段 → 平台 Lombok {@code @Data} 风格)。
 *
 * <p>改造点 (per 业务模块迁移流程 §七 Step 7):
 * <ul>
 *   <li>String id (csyh UUID 字符串) → Long id (雪花 ID, BaseEntity 已带)</li>
 *   <li>添加 tenant_id (P0-1 多租户拦截器)</li>
 *   <li>保留所有 csyh EnterprisePO 字段便于完整业务;Phase 1 控制器只暴露核心字段</li>
 *   <li>{@code @JsonFormat(shape=STRING)} on id 继承自 BaseEntity (Snowflake 精度保护)</li>
 * </ul>
 *
 * <p>关联关系:
 * <ul>
 *   <li>sys_enterprise_tag.enterprise_id → 本表.id (1:N)</li>
 *   <li>sys_enterprise_ent_bind.enterprise_id → 本表.id (1:N)</li>
 * </ul>
 *
 * <p>Phase 3 扩展预留:
 * <ul>
 *   <li>origin_id (天眼查) / is_sync (ES) / is_fill (云企库) 已建字段,逻辑延后</li>
 * </ul>
 *
 * @author Sisyphus (csyh 迁移)
 * @since 2026-06-30
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_enterprise")
public class Enterprise extends BaseEntity {

    private static final long serialVersionUID = 1L;

    /** 租户 ID (P0-1 多租户拦截器自动填充;@DataField 需配置 ignoreTable 或始终手动) */
    private Long tenantId;

    // ===== 基本信息 (Phase 1 核心字段) =====

    /** 企业名称 (必填) */
    private String name;

    /** 简称 */
    private String alias;

    /** 曾用名 */
    private String historyNames;

    /** 历史曾用名列表 (JSON/字符串) */
    private String historyNameList;

    /** 英文名 */
    private String engName;

    /** 纳税人识别号 */
    private String taxNumber;

    /** 统一社会信用代码 */
    private String creditCode;

    /** 行业 */
    private String industry;

    /** 国民经济行业分类门类 */
    private String category;

    /** 大类 */
    private String categoryBig;

    /** 中类 */
    private String categoryMiddle;

    /** 小类 */
    private String categorySmall;

    // ===== 法人 + 注册资本 =====

    /** 法人姓名 */
    private String legalPersonName;

    /** 法人类型 (1=人,2=公司) */
    private Integer type;

    /** 企业类型 */
    private String companyOrgType;

    /** 注册资本 */
    private String regCapital;

    /** 注册资本币种 (人民币/美元/欧元) */
    private String regCapitalCurrency;

    /** 实收资本 */
    private String actualCapital;

    /** 实收币种 */
    private String actualCapitalCurrency;

    /** 注册号 */
    private String regNumber;

    /** 组织机构代码 */
    private String orgNumber;

    // ===== 日期 =====

    /** 成立日期 */
    private LocalDateTime estiblishTime;

    /** 经营开始日期 */
    private LocalDateTime fromTime;

    /** 经营结束日期 */
    private LocalDateTime toTime;

    /** 核准时间 */
    private LocalDateTime approvedTime;

    /** 吊销日期 */
    private LocalDateTime revokeDate;

    /** 注销日期 */
    private LocalDateTime cancelDate;

    // ===== 注册地址 =====

    /** 省份 */
    private String base;

    /** 市 */
    private String city;

    /** 区 */
    private String district;

    /** 注册地址 */
    private String regLocation;

    /** 登记机关 */
    private String regInstitute;

    // ===== 经营状态 =====

    /** 企业状态 (在营/吊销/注销 等) */
    private String regStatus;

    /** 是否小微企业 (0=否 1=是) */
    private Integer isMicroEnt;

    /** 人员规模 (e.g. 100-499 人) */
    private String staffNumRange;

    /** 参保人数 */
    private Integer socialStaffNum;

    /** 经营范围 */
    private String businessScope;

    // ===== 联系信息 =====

    /** 联系电话 */
    private String phoneNumber;

    /** 邮箱 */
    private String email;

    /** 网站列表 */
    private String websiteList;

    // ===== 标签 + 评分 =====

    /** 标签列表 (冗余字段,主存 sys_enterprise_tag) */
    private String tags;

    /** 评分 */
    private Integer percentileScore;

    // ===== 股票相关 =====

    /** 股票号 */
    private String bondNum;

    /** 股票名 */
    private String bondName;

    /** 股票曾用名 */
    private String usedBondName;

    /** 股票类型 */
    private String bondType;

    // ===== 自身状态 =====

    /** 启用状态 (1=启用 0=停用) */
    private Integer status;

    // ===== 第三方关联 (Phase 3 占位) =====

    /** 企业 logo URL */
    private String logo;

    /** 天眼查 origin ID (csyh 保留字段) */
    private String originId;

    /** 是否已同步 ES (Phase 3) */
    private Integer isSync;

    /** 云企库同步状态 (0=无需 1=等待 2=完成) */
    private Integer isFill;

    // ===== BaseEntity 扩展审计字段 (Phase 1 平台约定) =====

    /** 创建人 loginName (BaseEntity 不带,需单独声明) */
    private String createBy;

    /** 更新人 loginName */
    private String updateBy;
}

File: code/platform-admin/src/api/enterprise.ts
import request from './request'

/**
 * 企业档案 API (park-enterprise 后端)
 * 后端端口: park-enterprise:8094
 * Gateway 路由: /enterprise/** → park-enterprise
 */

export interface Enterprise {
  id?: string
  name?: string          // 企业名称 (必填)
  alias?: string         // 简称
  historyNames?: string  // 曾用名
  historyNameList?: string
  engName?: string       // 英文名
  taxNumber?: string     // 纳税人识别号
  creditCode?: string    // 统一社会信用代码
  industry?: string      // 行业
  category?: string      // 国民经济行业分类
  categoryBig?: string   // 大类
  categoryMiddle?: string // 中类
  categorySmall?: string // 小类
  legalPersonName?: string  // 法人
  type?: number          // 法人类型
  companyOrgType?: string
  regCapital?: string    // 注册资本
  regCapitalCurrency?: string
  actualCapital?: string // 实收资本
  actualCapitalCurrency?: string
  regNumber?: string     // 注册号
  orgNumber?: string     // 组织机构代码
  estiblishTime?: string // 成立日期
  fromTime?: string
  toTime?: string
  approvedTime?: string
  revokeDate?: string
  cancelDate?: string
  base?: string          // 省份
  city?: string          // 市
  district?: string      // 区
  regLocation?: string   // 注册地址
  regInstitute?: string  // 登记机关
  regStatus?: string     // 经营状态
  isMicroEnt?: number
  staffNumRange?: string // 人员规模
  socialStaffNum?: number
  businessScope?: string // 经营范围
  phoneNumber?: string   // 联系电话
  email?: string
  websiteList?: string
  tags?: string          // 标签列表
  percentileScore?: number // 评分
  bondNum?: string
  bondName?: string
  usedBondName?: string
  bondType?: string
  logo?: string
  originId?: string
  isSync?: number
  isFill?: number
  tenantId?: number
  status?: number        // 1=启用 0=停用
  createTime?: string
  createBy?: string
  updateBy?: string
}

export function getEnterprisePage(params: {
  keyword?: string
  creditCode?: string
  industry?: string
  regStatus?: string
  hasBind?: number
  status?: number
  pageNum?: number
  pageSize?: number
}) {
  return request({ url: '/enterprise/page', method: 'get', params })
}

export function getEnterpriseById(id: string) {
  return request({ url: `/enterprise/${id}`, method: 'get' })
}

export function createEnterprise(data: Partial<Enterprise>) {
  return request({ url: '/enterprise', method: 'post', data })
}

export function updateEnterprise(id: string, data: Partial<Enterprise>) {
  return request({ url: `/enterprise/${id}`, method: 'put', data })
}

export function deleteEnterprise(id: string) {
  return request({ url: `/enterprise/${id}`, method: 'delete' })
}

export function toggleEnterpriseStatus(id: string, status: number) {
  return request({ url: `/enterprise/${id}/status`, method: 'patch', data: { status } })
}

// ==================== 批量操作 + Excel ====================

export function batchUpdateStatus(ids: string[], status: number) {
  return request({ url: '/enterprise/batch/status', method: 'put', data: { ids, status } })
}

export function batchDeleteEnterprise(ids: string[]) {
  return request({ url: '/enterprise/batch', method: 'delete', data: { ids } })
}

export function exportEnterpriseExcel(params?: any) {
  return request({ url: '/enterprise/excel/export', method: 'get', params, responseType: 'blob' })
}

export function downloadTemplateExcel() {
  return request({ url: '/enterprise/excel/template', method: 'get', responseType: 'blob' })
}

export function importEnterpriseExcel(file: File) {
  const formData = new FormData()
  formData.append('file', file)
  return request({ url: '/enterprise/excel/import', method: 'post', data: formData, headers: { 'Content-Type': 'multipart/form-data' } })
}

// ==================== V54: 企业行业类型 API ====================

export interface EnterpriseIndustry {
  id?: string
  tenantId?: number
  code?: string         // 行业代码 (GB/T 4754-2017)
  category?: string     // 门类
  categoryBig?: string  // 大类
  categoryMiddle?: string
  categorySmall?: string
  status?: number       // 1=启用 0=停用
  createTime?: string
}

export function getIndustryPage(params: {
  current?: number
  size?: number
  keyword?: string
  category?: string
  status?: number
}) {
  return request({ url: '/enterprise/industry/page', method: 'get', params })
}

export function getIndustryById(id: string) {
  return request({ url: `/enterprise/industry/${id}`, method: 'get' })
}

export function createIndustry(data: Partial<EnterpriseIndustry>) {
  return request({ url: '/enterprise/industry', method: 'post', data })
}

export function updateIndustry(id: string, data: Partial<EnterpriseIndustry>) {
  return request({ url: `/enterprise/industry/${id}`, method: 'put', data })
}

export function deleteIndustry(id: string) {
  return request({ url: `/enterprise/industry/${id}`, method: 'delete' })
}

// ==================== V56: 企业评分规则 API ====================

export interface Rating {
  id?: string
  tenantId?: number
  parkId?: number
  level?: number        // 1=优 2=良 3=中 4=差
  overdueMin?: number
  overdueMax?: number
  debtsMin?: number
  debtsMax?: number
  dateNum?: number
  dateUnit?: string     // day/month/year
  status?: number
}

export function getRatingList(tenantId?: number) {
  return request({ url: '/enterprise/rating/list', method: 'get', params: { tenantId } })
}

export function saveRatingBatch(ratings: Rating[]) {
  return request({ url: '/enterprise/rating/save', method: 'put', data: ratings })
}

export function resetRatingDefault(tenantId?: number) {
  return request({ url: '/enterprise/rating/reset', method: 'post', params: { tenantId } })
}

// ==================== V58: 客户信息 API ====================

export interface CustomerInformation {
  id?: string
  enterpriseId?: number
  code?: string
  customerType?: number        // 1=潜在 2=意向 3=已签约
  area?: number
  settleAddress?: string
  url?: string
  name?: string                // 负责人
  phone?: string
  managerPhone?: string
  email?: string
  registTime?: string
  registMoney?: number
  rentalStandard?: string
  industrialField?: number      // 1-6
  industrialFieldOther?: string
  mainBusiness?: string
  companyStrengths?: string     // 多选 1-10, 逗号分隔
  companyStrengthsOther?: string
  validIntellectualProperty?: number
  inventionPatents?: number
  utilityModelPatent?: number
  industrialDesignPatents?: number
  trademark?: number
  softwareCopyright?: number
  newPlantVariety?: number
  integratedCircuitLayout?: number
  purchaseForeignPatents?: number
  otherPatents?: number
  businessIncome?: number
  lastYearTax?: number
  totalInvestmentAmount?: number
  totalFinancingAmount?: number
  companyDifficulties?: string
  companyDifficultiesOther?: string
  supportingServices?: string
  technicalConsultingServices?: string
  managementServices?: string
  managementServicesOther?: string
  technologyPlatformServices?: string
  technologyPlatformServicesOther?: string
  investmentServices?: string
  investmentServicesOther?: string
  suggest?: string
  structuralLoad?: number
  floorHeight?: number
  capacitance?: number
  supplyAndDrainage?: string
  freshAirSmokeExhaust?: number
  elevatorLength?: number
  elevatorWidth?: number
  elevatorHeight?: number
  elevatorLoad?: number
  status?: number
}

export function getCustomerPage(params: {
  current?: number
  size?: number
  customerType?: number
  status?: number
}) {
  return request({ url: '/enterprise/customer/page', method: 'get', params })
}

export function getCustomerById(id: string) {
  return request({ url: `/enterprise/customer/${id}`, method: 'get' })
}

export function createCustomer(data: Partial<CustomerInformation>) {
  return request({ url: '/enterprise/customer', method: 'post', data })
}

export function updateCustomer(id: string, data: Partial<CustomerInformation>) {
  return request({ url: `/enterprise/customer/${id}`, method: 'put', data })
}

export function deleteCustomer(id: string) {
  return request({ url: `/enterprise/customer/${id}`, method: 'delete' })
}

// ==================== V57: 关注标签 + 内容 API ====================

export interface Focus {
  id?: string
  name?: string
  sorting?: number
  status?: number
}

export interface FocusItem {
  id?: string
  focusId?: number
  name?: string
  sorting?: number
  status?: number
}

export function getFocusPage(params: {
  current?: number
  size?: number
  keyword?: string
  status?: number
}) {
  return request({ url: '/enterprise/focus/page', method: 'get', params })
}

export function createFocus(data: Partial<Focus>) {
  return request({ url: '/enterprise/focus', method: 'post', data })
}

export function updateFocus(id: string, data: Partial<Focus>) {
  return request({ url: `/enterprise/focus/${id}`, method: 'put', data })
}

export function deleteFocus(id: string) {
  return request({ url: `/enterprise/focus/${id}`, method: 'delete' })
}

export function getFocusItemByFocusId(focusId: number | string) {
  return request({ url: `/enterprise/focus-item/by-focus/${focusId}`, method: 'get' })
}

export function createFocusItem(data: Partial<FocusItem>) {
  return request({ url: '/enterprise/focus-item', method: 'post', data })
}

export function updateFocusItem(id: string, data: Partial<FocusItem>) {
  return request({ url: `/enterprise/focus-item/${id}`, method: 'put', data })
}

export function deleteFocusItem(id: string) {
  return request({ url: `/enterprise/focus-item/${id}`, method: 'delete' })
}
